docs/CLOUD_RECORDING.md

```diff
@@ -0,0 +1,284 @@
+
+The inputs will need the `method` (we will use GET and POST), `url`, `body`, and the `credential`. 
+
+This credential will be generated by another helper function in `utils/generateCredential.ts`. This is a base-64 encoded credential using the Customer ID and Customer Secret.
+
+```ts
+export const generateCredential = () => {
+  const credential = import.meta.env.CUSTOMER_ID + ":" + import.meta.env.CUSTOMER_SECRET
+
+  const base64_credential = btoa(credential)
+  return base64_credential
+}
+```
+
+The `makeRequest` function defines all the headers, executes the request, and returns the response. 
+
+```ts
+export const makeRequest = async (method: string, url: string, credential: string, body?: string) => {
+    const headers = new Headers({
+        "Authorization": "basic " + credential,
+        "Content-Type": "application/json",
+        "Access-Control-Allow-Origin": "*",
+        "Access-Control-Allow-Methods": `${method}, OPTIONS`,
+    });
+
+    const res = await fetch(url, {
+        method: method,
+        headers: headers,
+        body: body
+    })
+    if (!res.ok) {
+        console.log(await res.text())
+        throw new Error("Failed to make request")
+    }
+    return res
+}
+```
+
+## Generate Resource
+The first step in the cloud recording process is to generate a resource for cloud recording. Behind the scenes this spins up a backend Agora service that has the capability of recording your video calls. 
+
+> Your cloud recording needs to be started within five minutes, or else the resource will shut down.
+
+To generate this resource, we make our first request to the Agora service. This will use the `acquire` function. You must pass the channel name and the uid for the resource. 
+
+This function will only be called when a video is being started, so there is no reason to expose an endpoint to generate a resource by itself. To keep the code organized this function will live in the `utils/generateResource.ts` file. 
+
+```ts
+import { makeRequest } from "./makeRequest"
+
+export const generateCloudRecordingResource = async (channel: string, credential: string, uid: string, appId: string) => {
+
+    const payload = {
+        "cname": channel,
+        "uid": uid,
+        "clientRequest": {}
+    }
+
+    const url = `https://api.agora.io/v1/apps/${appId}/cloud_recording/acquire`
+
+    const res = await makeRequest("POST", url, credential, JSON.stringify(payload))
+    const data = await res.json()
+    const resourceId = data["resourceId"]
+
+    return resourceId
+}
+```
+
+## Start Cloud Recording
+Now we can put it all together and start the recording within the POST request in the `api/recording/start.json.ts` file. We generate the credential, then use that credential to generate a resource, and create a token.
+
+Then we make a request with the `start` endpoint url and pass in the payload. This payload is where we define all the information that the Agora cloud recording service needs.
```

Comment:
Missing context, what start url (yours or agora)? What's the payload, do you mean the request body?

Reply:
Updated to mention agora, and use request body instead of payload